### PR TITLE
[MI-2280]:Add Dynamic Argument for config add command

### PR DESCRIPTION
**Pull request by Kshitij-Katiyar**

### CODE REVIEW

File: server/instance_list.go [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+	"github.com/mattermost/mattermost-server/v6/model"
+)
+
+func (p *Plugin) handleGetInstanceList(w http.ResponseWriter, r *http.Request) {
+	instances, err := p.instanceStore.LoadInstances()
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+	out := []model.AutocompleteListItem{}
```
Comment (ayusht2810) [inline server/instance_list.go]:
```suggestion
	}

	out := []model.AutocompleteListItem{}
```

File: server/instance_list.go [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+	for _, ID := range instances.IDs() {
+		out = append(out, model.AutocompleteListItem{
+			Item: ID.String(),
+		})
+	}
+	b, _ := json.Marshal(out)
```
Comment (ayusht2810) [inline server/instance_list.go]:
```suggestion
	}

	b, _ := json.Marshal(out)
```

Comment (ayusht2810) [reply server/instance_list.go]: give proper name

**Review by raghavaggarwal2308 — APPROVED** [review]

**Review by ayusht2810 — APPROVED** [review]

File: server/instance_list.go [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+	}
+
+	out := []model.AutocompleteListItem{}
+	for _, ID := range instances.IDs() {
+		out = append(out, model.AutocompleteListItem{
+			Item: ID.String(),
+		})
+	}
+
+	response, _ := json.Marshal(out)
```
Comment (manojmalik20) [inline server/instance_list.go:24]: Why not check the error here?

**Review by manojmalik20 — APPROVED** [review]

File: server/config_list.go [hunk 4]
```diff
@@ -34,7 +34,11 @@ func (p *Plugin) handleGetConfigList(w http.ResponseWriter, r *http.Request) {
 		})
 	}
 
-	response, _ := json.Marshal(out)
+	response, err := json.Marshal(out)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusUnauthorized)
```
Comment (manojmalik20) [inline server/config_list.go]:
```suggestion
		http.Error(w, err.Error(), http.StatusInternalServerError)
```

**Review by manojmalik20 — APPROVED** [review]

**Pull request merged**